android/java/org/chromium/chrome/browser/rate/BraveRateThanksFeedbackDialog.java

```diff
@@ -0,0 +1,108 @@
+
+package org.chromium.chrome.browser.rate;
+
+import android.annotation.SuppressLint;
+import android.app.Dialog;
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.Button;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+
+import org.chromium.base.task.AsyncTask;
+import org.chromium.chrome.R;
+
+public class BraveRateThanksFeedbackDialog extends BottomSheetDialogFragment {
+    final public static String TAG_FRAGMENT = "brave_rate_thanks_feedback_dialog_tag";
+    private static final String TAG = "ask_play_store";
+    private static final String SAD = "sad";
+
+    public static BraveRateThanksFeedbackDialog newInstance() {
+        return new BraveRateThanksFeedbackDialog();
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setStyle(STYLE_NORMAL, R.style.AppBottomSheetDialogTheme);
+    }
+
+    @Override
+    public void show(@NonNull FragmentManager manager, @Nullable String tag) {
+        try {
+            BraveRateThanksFeedbackDialog fragment =
+                    (BraveRateThanksFeedbackDialog) manager.findFragmentByTag(
+                            BraveRateThanksFeedbackDialog.TAG_FRAGMENT);
+            FragmentTransaction transaction = manager.beginTransaction();
+            if (fragment != null) {
+                transaction.remove(fragment);
+            }
+            transaction.add(this, tag);
+            transaction.commitAllowingStateLoss();
+        } catch (IllegalStateException e) {
+            Log.e("BraveRateDialogFragment", e.getMessage());
+        }
+    }
+
+    @SuppressLint("RestrictedApi")
+    @Override
+    public void setupDialog(@NonNull Dialog dialog, int style) {
+        super.setupDialog(dialog, style);
+
+        final View view = LayoutInflater.from(getContext())
+                                  .inflate(R.layout.brave_rating_thanks_feedback_dialog, null);
+        addSuggestionEditText(view);
+        clickOnDoneButton(view);
+        dialog.setContentView(view);
+    }
+
+    private void clickOnDoneButton(View view) {
+        Button doneButton = view.findViewById(R.id.rate_done_button);
+        doneButton.setOnClickListener((v) -> {
+            TextInputEditText feedbackEditText = view.findViewById(R.id.feedbackEditText);
+            String feedBack = feedbackEditText.getText().toString();
+            RateFeedbackUtils.RateFeedbackWorkerTask mWorkerTask =
```

Comment:
why do we call a local variable as a class member `mWorkerTask`? It's confusing.

Reply:
updated